File: biweekly/098/q2.cc
class Solution {
public:
    int minimizeSum(vector<int>& nums) {
        int n = nums.size();
        if (n == 3) return 0;
        sort(nums.begin(), nums.end());
        return min({nums.back() - nums[2], *(nums.end() - 3) - nums[0], *(nums.end() - 2) - nums[1]});
    }
};

File: weekly/350_/q2.go
func findValueOfPartition(nums []int) int {
    sort.Ints(nums)
    ans := math.MaxInt
    n := len(nums)
    for i := 0; i < n - 1; i++ {
        ans = min(ans, abs(nums[i] - nums[i + 1]))
    }
    return ans
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

